### PR TITLE
Implemented controller to Event

### PR DESCRIPTION
**Pull request by Dunaysky**
Implemented controller to Event, added serializer and spesc.

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def destroy
+    event.delete
+  end
+
+  private
+
+  def event_params
+    params.require(:event).permit(:title, :description, :date)
```
Comment (danyil-cybercraft) [inline app/controllers/events_controller.rb]: Please add `user_ids` to params to be able to add users to this event

File: app/serializers/event_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class EventSerializer
+  include JSONAPI::Serializer
+  attributes :id
+  attribute :eventTitle, &:title
```
Comment (danyil-cybercraft) [inline app/serializers/event_serializer.rb]: You should not change title key name, just leave title. The same for description

File: spec/requests/event_spec.rb [hunk 3]
```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Event' do
+  describe 'POST /events' do
```
Comment (danyil-cybercraft) [inline spec/requests/event_spec.rb]: Please split positive and negative test cases into defferent context blocks

File: spec/requests/event_spec.rb [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+      expect(response).to have_http_status(:unprocessable_entity)
+    end
+  end
+
+  describe 'GET /enents/:event_id' do
+    let!(:event) { create(:event) }
+
+    it 'returns json representation of the event' do
+      get "/events/#{event.id}"
```
Comment (danyil-cybercraft) [inline spec/requests/event_spec.rb]: You can put this in before block instead of calling in all tests

File: spec/requests/event_spec.rb [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    it 'returns more than one event' do
+      get '/events'
+      expect(response).to have_http_status(:ok)
+      expect(json['data']&.count).to eq(number_of_events)
+    end
+
+    it 'returns properly serialized events' do
+      get '/events'
+
+      event_to_compare = json['data'].find do |event|
```
Comment (danyil-cybercraft) [inline spec/requests/event_spec.rb]: You can move this in let block

File: spec/requests/event_spec.rb (hunk 4 above)
Comment (Dunaysky) [reply spec/requests/event_spec.rb]: If I use before block:
```
before do
      @event = create(:event)
    end

    it 'returns json representation of the event' do
      get "/events/#{@event.id}"
...
```
the rubocop don't pass and has offenses:
```
spec/requests/event_spec.rb:67:22: C: RSpec/InstanceVariable: Avoid instance variables – use let, a method call, or a local variable (if possible).
      get "/events/#{@event.id}"
                     ^^^^^^
```
So leave it in let block.

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (Dunaysky) [reply app/controllers/events_controller.rb]: Added `user_ids`
```
  def event_params
    params.require(:event).permit(:title, :description, :date, user_ids: [])
  end

```

File: app/serializers/event_serializer.rb (hunk 2 above)
Comment (Dunaysky) [reply app/serializers/event_serializer.rb]: Serialized only date:
```
  attributes :id, :title, :description
  attribute :eventDate do |object|
    object.date.to_fs(:short)

```

File: spec/requests/event_spec.rb (hunk 3 above)
Comment (Dunaysky) [reply spec/requests/event_spec.rb]: Splitted into different context blocks

File: spec/requests/event_spec.rb (hunk 5 above)
Comment (Dunaysky) [reply spec/requests/event_spec.rb]: Moved to the let block:
```
    let(:event_to_compare) do
      json['data'].find do |event|
        event['id'].to_i == first_event.id
      end
    end

```

File: app/controllers/events_controller.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class EventsController < ApplicationController
```
Comment (danyil-cybercraft) [inline app/controllers/events_controller.rb]: EventsController should be in `api/v1` folder

File: config/routes.rb [hunk 7]
```diff
@@ -3,4 +3,5 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  resources :events, only: %i[index create show update destroy]
```
Comment (danyil-cybercraft) [inline config/routes.rb]: Events recource should be in `api/v1` namespace

File: app/controllers/events_controller.rb (hunk 6 above)
Comment (Dunaysky) [reply app/controllers/events_controller.rb]: Put controller to `api/v1` folder and added `module Api` and `module V1`

File: config/routes.rb (hunk 7 above)
Comment (Dunaysky) [reply config/routes.rb]: Added namespace `api` and `v1`

**Review by danyil-cybercraft — APPROVED** [review]

**Pull request merged**